tests/unittests/Foundation/ReferenceFoundation/TestNSArray.mm

```diff
@@ -150,13 +150,19 @@ static int objectIndexInArray(NSArray* array, int value, int startingFrom, int l
     //        NSBinarySearchingOptions* searchForBoth = NSBinarySearchingFirstEqual | NSBinarySearchingLastEqual
     //        auto _ = objectIndexInArray(array, 9, 0, 13, searchForBoth)
 
-    auto notFound = objectIndexInArray(array, 11, 0, 13);
-    ASSERT_EQ_MSG(notFound, NSNotFound, @"NSArray return NSNotFound if object is not found.");
+    ASSERT_EQ_MSG(objectIndexInArray(array, 11, 0, array.count),
```

Comment:
The test on a subrange is done after these 3 tests. In fact, there was a bug in the current version of the algorithm where an exception was thrown only when one was searching in the whole array. The name of the variable was let me think that the developer who wrote the test wanted to search on the whole range for this case, and on a subrange for the other test (which has been kept). Unfortunately, the original test was not able to trigger the issue, because of a wrong index. Now the tests cover both cases.